lib/auth/clt.go

```diff
@@ -524,6 +524,26 @@ func (c *Client) IntegrationAWSOIDCClient() integrationv1.AWSOIDCServiceClient {
 	return integrationv1.NewAWSOIDCServiceClient(c.APIClient.GetConnection())
 }
 
+// ListRemoteClusters returns a page of remote clusters.
```

Comment:
These two feel very weird - but without making major changes to a heavily shared interface, they seem unavoidable. Thoughts ?

Reply:
Actually - now that I think about it our hands might be tied here. Even if I split the interfaces, when it comes to the next PR which will add the gRPC methods, for the purposes of back and forward compatibility, we'll need this behaviour in the legacy client anyway.

Reply:
Spent an hour or so playing with this and I think your solution in the Users refactor which is what I've gone for here is correct. Unfortunately, there's one place we need the Client implementation and the Backend implementation to share a signature: the cache. I think that's a discussion/investigation for another day and it's safe enough to follow the same route you've gone with for the users service here.